buildkite/scripts/version-linter-patch-missing-type-shapes.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -eox pipefail
+
+if [[ $# -ne 1 ]]; then
+    echo "Usage: $0 <release-branch>"
+    exit 1
+fi
+
+git config --global --add safe.directory /workdir
+
+source buildkite/scripts/handle-fork.sh
+source buildkite/scripts/export-git-env-vars.sh
+
+release_branch=${REMOTE}/$1
+
+RELEASE_BRANCH_COMMIT=$(git log -n 1 --format="%h" --abbrev=7 $release_branch)
+
+function revert_checkout() {
+    git checkout $BUILDKITE_COMMIT
+    git submodule sync
+    git submodule update --init --recursive
+}
+
+function checkout_and_dump() {
+    local __commit=$1
+    git checkout $__commit
+    git submodule sync
+    git submodule update --init --recursive
+    eval $(opam config env)
+    TYPE_SHAPE_FILE=${__commit:0:7}-type_shape.txt
+    dune exec src/app/cli/src/mina.exe internal dump-type-shapes > /tmp/${TYPE_SHAPE_FILE}
+    revert_checkout
+    source buildkite/scripts/gsutil-upload.sh /tmp/${TYPE_SHAPE_FILE} gs://mina-type-shapes
+}
+
+if ! $(gsutil ls gs://mina-type-shapes/$BUILDKITE_COMMIT 2>/dev/null); then
```

Comment:
`BUILDKITE_COMMIT` and `RELEASE_BRANCH_COMMIT` are treated uniformly here, however they seem to have different semantics.

`RELEASE_BRANCH_COMMIT` is abbreviated to 7 symbols, and hence using it in file name is what one would expect.

On the other hand, `gsutil ls gs://mina-type-shapes/$BUILDKITE_COMMIT` will always fail, because the `BUILDKITE_COMMIT` contains full commit name, AFAIU

I suggest adding code to abbreviate `BUILDKITE_COMMIT` to 7 symbols as well.

After that the code above could be simplified as well:

```
TYPE_SHAPE_FILE=${__commit:0:7}-type_shape.txt
```

to

```
TYPE_SHAPE_FILE=$__commit-type_shape.txt
```

Reply:
nice catch !